V_NOP | PfpSyncMe | S_CMPK_EQ_U32

DanielSvoboda opened the pull request:
`[Render.Recompiler] <Error> translate.cpp:LogMissingOpcode:443: Unknown opcode V_NOP (463, category = VectorALU)`
V_NOP = Do nothing

`[Render.Recompiler] <Error> translate.cpp:LogMissingOpcode:443: Unknown opcode S_CMPK_EQ_U32 (54, category = ScalarALU)`
SCC = (D.u == SIMM16)

`
[Debug] <Critical> liverpool.cpp:ProcessGraphics:446: Unreachable code!
Unknown PM4 type 3 opcode 0x42 with count 1
`
0x42 = PfpSyncMe


`[Render.Vulkan] <Error> image_view.cpp:ImageViewInfo:109: Storage image (num_comps = 4) requires swizzling [BGRA]`
format 43 dst_sel 3886

DanielSvoboda committed:
V_NOP

V_NOP = Do nothing

DanielSvoboda committed:
PfpSyncMe

PfpSyncMe ensures that all previous commands are completed before continuing.
'break' should be enough for now

DanielSvoboda committed:
S_CMPK_EQ_U32

S_CMPK_EQ_U32
SCC = (D.u == SIMM16)

DanielSvoboda committed:
S_CMPK_EQ_U32

DanielSvoboda committed:
OperandField::Undefined:

raphaelthegreat commented on src/shader_recompiler/frontend/translate/translate.cpp:
```diff
@@ -149,6 +149,8 @@ T Translator::GetSrc(const InstOperand& operand) {
         } else {
             return m0_value;
         }
+    case OperandField::Undefined:
```
What is the purpose of this

DanielSvoboda replied:
There was a game that went to 'UNREACHABLE' because it was called that, but it was in the middle of several modifications, maybe I shouldn't include that

DanielSvoboda committed:
Update translate.cpp

remove  OperandField::Undefined:

DanielSvoboda committed:
Update image_view.cpp

[Render.Vulkan] <Error> image_view.cpp:ImageViewInfo:109: Storage image (num_comps = 4) requires swizzling [BGRA]
format 43 dst_sel 3886

DanielSvoboda committed:
Update liverpool_to_vk.cpp

raphaelthegreat commented on src/shader_recompiler/frontend/translate/scalar_alu.cpp:
```diff
@@ -463,4 +465,10 @@ void Translator::S_MIN_U32(const GcnInst& inst) {
 }
 
+void Translator::S_CMPK_EQ_U32(const GcnInst& inst) {
+    const IR::U32 src0{GetSrc(inst.src[0])};
```
Better use simm16 here manually like S_MOVK

DanielSvoboda replied:
ok

DanielSvoboda committed:
S_CMPK_EQ_U32

DanielSvoboda committed:
S_CMPK_EQ_U32
